packages/use-dataloader/src/dataloader.ts

```diff
@@ -1,248 +1,152 @@
 import { DEFAULT_MAX_CONCURRENT_REQUESTS, StatusEnum } from './constants'
-import {
-  NeedPollingType,
-  OnCancelFn,
-  OnErrorFn,
-  OnSuccessFn,
-  PromiseType,
-} from './types'
+import { PromiseType } from './types'
 
 export type DataLoaderConstructorArgs<T = unknown> = {
   key: string
   method: () => PromiseType<T>
-  pollingInterval?: number
-  needPolling?: NeedPollingType<T>
-  maxDataLifetime?: number
-  keepPreviousData?: boolean
+  enabled?: boolean
+  notifyChanges?: () => void
 }
 
-class DataLoader<T = unknown> {
-  public static started = 0
-
+class DataLoader<ResultType = unknown, ErrorType = unknown> {
   public static maxConcurrent = DEFAULT_MAX_CONCURRENT_REQUESTS
 
-  public static cachedData = {} as Record<string, unknown | undefined>
-
-  public key: string
-
-  public status: StatusEnum
-
-  public pollingInterval?: number
-
-  public needPolling: NeedPollingType<T> = true
-
-  public maxDataLifetime?: number
-
-  public isDataOutdated = false
+  public static started = 0
 
-  public method: () => PromiseType<T>
+  public static cachedData = {} as Record<string, unknown | undefined>
 
-  private cancelMethod?: () => void
+  public static queue = {} as Record<string, PromiseType>
```

Comment:
Any reason why casting instead of normally typing these variables?

Reply:
No lol